tests/src/test/scala/scalaz/IListTest.scala

```diff
@@ -409,6 +409,16 @@ object IListTest extends SpecLite {
     ns.zipWithIndex.toList must_=== ns.toList.zipWithIndex
   }
 
+  "IList.:: pattern match" ! forAll { (n: Int, ns: IList[Int]) =>
+    import IList.::
```

Comment:
Would you not want this available without the explicit import?

Reply:
it would conflict with the stdlib `::`, and also shapeless, so at this point I think opt-in is the best way to go. Maybe in scalaz 8 :wink: